File: microservices/hrm-service/src/main/java/tn/iteam/backend/service/PayrollService.java
package tn.iteam.backend.service;

import java.util.List;
import tn.iteam.backend.entity.Payroll;

public interface PayrollService {
    List<Payroll> findVisible();

    List<Payroll> findForUser(Long userId);

    Payroll findById(Long id);

    Payroll save(Payroll payroll);

    Payroll update(Long id, Payroll payroll);

    void delete(Long id);
}

File: frontend/src/app/services/payroll.service.ts
import { inject, Injectable } from '@angular/core';
import { ApiService } from './api.service';
import type { Payroll } from '../models/payroll.model';

@Injectable({ providedIn: 'root' })
export class PayrollService {
  private readonly api = inject(ApiService);

  list() {
    return this.api.client.get<Payroll[]>(`${this.api.baseUrl}/payroll`);
  }

  forUser(userId: number) {
    return this.api.client.get<Payroll[]>(`${this.api.baseUrl}/payroll/user/${userId}`);
  }

  get(id: number) {
    return this.api.client.get<Payroll>(`${this.api.baseUrl}/payroll/${id}`);
  }

  create(body: Payroll) {
    return this.api.client.post<Payroll>(`${this.api.baseUrl}/payroll`, body);
  }

  update(id: number, body: Payroll) {
    return this.api.client.put<Payroll>(`${this.api.baseUrl}/payroll/${id}`, body);
  }

  delete(id: number) {
    return this.api.client.delete(`${this.api.baseUrl}/payroll/${id}`);
  }
}
